src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/DoNotGuardDictionaryRemoveByContainsKeyTests.cs

```diff
@@ -391,6 +393,29 @@ End Class
 
             await VerifyVB.VerifyAnalyzerAsync(source);
         }
+
+        [Fact]
+        [WorkItem(6377, "https://github.com/dotnet/roslyn-analyzers/issues/6377")]
+        public async Task ContainsKeyIsCalledOnDifferentInstance_NoDiagnostic_CS()
```

Comment:
💡 This only tests fields. Can you also test the following:

1. Properties
2. Locals
3. Include a negated condition (`!a.ContainsKey`) to make sure the case described in #6377 is tested explicitly

Reply:
You're absolutely right! I was a bit too hasty in undrafting this PR... Sorry.

Reply:
Ok so I made the tests a little more comprehensive, but I'll gladly add more if need be.